src/core/load-fragment.c

```diff
@@ -2725,17 +2726,18 @@ int config_parse_syscall_filter(
                         c->syscall_whitelist = true;
 
                         /* Accept default syscalls if we are on a whitelist */
-                        r = syscall_filter_parse_one(unit, filename, line, c, false, "@default", false);
+                        r = syscall_filter_parse_one(unit, filename, line, c, false, "@default", false, -1);
                         if (r < 0)
                                 return r;
                 }
         }
 
         p = rvalue;
         for (;;) {
-                _cleanup_free_ char *word = NULL;
+                _cleanup_free_ char *name = NULL, *e = NULL;
+                int num = -1;
 
-                r = extract_first_word(&p, &word, NULL, 0);
+                r = extract_first_word(&p, &name, ":" WHITESPACE, 0);
```

Comment:
... and this is a second one.

I think it'd make sense to provide a function like `parse_syscall_and_errno(const char *p, char **p, int *error)`. This way the code would be a bit clearer, and there'd be only one parser. It'll also be easier to add a unit test for the parser.

Reply:
OK. `parse_syscall_and_errno()` is added.